pkg/remediator/reconcile/reconciler.go

```diff
@@ -168,6 +169,29 @@ func (r *reconciler) remediate(ctx context.Context, id core.ID, objDiff diff.Dif
 		}
 		klog.V(3).Infof("Remediator abandoning object %v", id)
 		return r.applier.RemoveNomosMeta(ctx, actual, metrics.RemediatorController)
+	case diff.UpdateCSMetadata:
+
+		actual, err := objDiff.UnstructuredActual()
+		if err != nil {
+			return err
+		}
+
+		uObj := &unstructured.Unstructured{}
+
+		uObj.SetGroupVersionKind(objDiff.Declared.GetObjectKind().GroupVersionKind())
+		uObj.SetName(objDiff.Declared.GetName())
+		csAnnotations, csLabels := metadata.GetConfigSyncMetadata(objDiff.Declared)
+
+		uObj.SetAnnotations(objDiff.Actual.GetAnnotations())
+		uObj.SetLabels(objDiff.Actual.GetLabels())
+		metadata.RemoveConfigSyncMetadata(uObj)
+		core.RemoveAnnotations(uObj, metadata.LifecycleMutationAnnotation)
```

Comment:
Do we need to remove the LifecycleMutationAnnotation explicitly? csAnnotations should reset it.

Reply:
If Actual has the LifecycleMutationAnnotation but Declared doesn't, then uObj will have the LifecycleMutationAnnotation. That's why it's needed